test/01_contraction/contraction_test.cpp

```diff
@@ -367,6 +395,48 @@ namespace hiptensor
                     hiptensorPrintArrayElements<float>(stream, (float*)D.get(), elementsCD);
                     stream << std::endl;
                 }
+                else if(DDataType == HIP_C_32F)
+                {
+                    stream << "Tensor A elements:\n";
```

Comment:
The type of D shouldn't determine the type of the rest of ABC.
What we need to do here is implement a wrapper:
```
hipTensorPrintArrayElements(stream, void* data, elementCount, hipDataType dataType)
{
    if (dataType == HIP_R_32F)
    {
        hipTensorPrintArrayElements<float>(...);
    } else if...
}
```

That way the dispatching is wrapped up. The new code should look something like
```
hipTensorPrintArrayElements(stream, resource->hostA().get(), elementsA, ADataType);
hipTensorPrintArrayElements(stream, resource->hostB().get(), elementsA, BDataType);
...
```